cider/exec/template/InValuesIR.cpp

```diff
@@ -126,6 +129,59 @@ std::unique_ptr<CodegenColValues> CodeGenerator::codegenInValues(
   return std::make_unique<FixedSizeColValues>(result, lhs_null);
 }
 
+std::unique_ptr<CodegenColValues> CodeGenerator::codegenInValuesString(
+    const Analyzer::InValues* expr,
+    const CompilationOptions& co) {
+  const auto in_arg = expr->get_arg();
+  const auto& expr_ti = expr->get_type_info();
+  CHECK(expr_ti.is_boolean());
+  auto lhs_lv = codegen(in_arg, co, true);
+  auto lhs_fixsize = dynamic_cast<TwoValueColValues*>(lhs_lv.get());
+  CHECK(lhs_fixsize);
+  auto lhs_str_ptr = lhs_fixsize->getValueAt(0);
+  auto lhs_str_len = lhs_fixsize->getValueAt(1);
+  auto lhs_null = lhs_fixsize->getNull();
+  llvm::Value* result{nullptr};
+  if (expr_ti.get_notnull()) {
+    result = llvm::ConstantInt::get(llvm::IntegerType::getInt1Ty(cgen_state_->context_),
+                                    false);
+  } else {
+    result = cgen_state_->llBool(false);
+  }
+  CHECK(result);
+
+  if (co.hoist_literals) {  // TODO(alex): remove this constraint
+    auto in_vals_bitmap = createInValuesBitmapArrow(expr, co);
+    if (in_vals_bitmap) {
+      if (in_vals_bitmap->isEmpty()) {
+        return in_vals_bitmap->hasNull()
+                   ? std::make_unique<FixedSizeColValues>(
+                         cgen_state_->inlineIntNull(SQLTypeInfo(kBOOLEAN, false)),
+                         lhs_null)
+                   : std::make_unique<FixedSizeColValues>(result, lhs_null);
+      }
+      const int64_t cider_string_hasher_handle =
+          reinterpret_cast<int64_t>(executor()->getCiderStringHasherHandle());
+      auto cider_string_hasher_handle_lv = cgen_state_->llInt(cider_string_hasher_handle);
+      auto id_val = cgen_state_->emitCall(
+          "look_up_string_id_from_hasher",
+          {lhs_str_ptr, lhs_str_len, cider_string_hasher_handle_lv});
+      return cgen_state_->addInValuesBitmap(in_vals_bitmap)
+          ->codegen(id_val, lhs_null, executor());
+    } else {
+      LOG(INFO) << "Bitmap not created, switch to using OR.";
+    }
+  }
+
+  for (auto in_val : expr->get_value_list()) {
```

Comment:
Here we also need to cover non-string case like integer so can't just call `codegenStringEq`.